### DISCUSSION TITLE
API Github - Search repositories - sort updated - Format timestamp

### DISCUSSION THREAD

**Discussion by gabrieldeoliveiraestevam** [community/community#24746] [Apps, API and Webhooks]
<p>Guys,</p>
<p>I need to make a request ordering by “updated” and that the “updated” is greater than X.</p>
<p>I tried to put in place of “X” several types of timestamp / date and none got success.</p>
<p>Format:</p>
<p>“2019-03-31T00:45:57Z”</p>
<p>2019-03-31T00:45:57Z</p>
<p>31/03/2019</p>
<p>31.03.2019</p>
<p>Request:</p>
<pre><code class="lang-auto">https://api.github.com/search/repositories?q=topic:data-science+updated:&gt;=X&amp;sort=updated&amp;order=desc&amp;page=1&amp;per_page=100
</code></pre>
<p>The documentation shows that I can sort by “updated”, but it doesn’t show how to search for “updated”.</p>
<blockquote>
<p><a href="https://docs.github.com/en/rest/reference/search#search-repositories" class="inline-onebox" rel="noopener nofollow ugc">Search - GitHub Docs</a></p>
</blockquote>

**Comment by parithiban** [comment, marked as answer]
<p><a class="mention" href="/gabrieldeoliveiraestevam">@gabrieldeoliveiraestevam</a></p>
<p>Kindly have a look at this</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-for-repositories#search-by-when-a-repository-was-created-or-last-updated" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-for-repositories#search-by-when-a-repository-was-created-or-last-updated" target="_blank" rel="noopener nofollow ugc">Searching for repositories - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-for-repositories#search-by-when-a-repository-was-created-or-last-updated" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-for-repositories</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/github/searching-for-information-on-github/getting-started-with-searching-on-github/understanding-the-search-syntax#query-for-dates" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/github/searching-for-information-on-github/getting-started-with-searching-on-github/understanding-the-search-syntax#query-for-dates" target="_blank" rel="noopener nofollow ugc">Understanding the search syntax - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/github/searching-for-information-on-github/getting-started-with-searching-on-github/understanding-the-search-syntax#query-for-dates" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/github/searching-for-information-on-github/getting-started-with-searching-on-github/understanding-the-search-syntax</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>If you are using repository type in a search query you can use <code>pushed</code> instead of <code>updated</code>.</p>
<p>The query which you were expecting should be something like this</p>
<pre><code class="lang-auto">https://api.github.com/search/repositories?q=topic:data-science+pushed:&gt;2019-03-31&amp;sort=updated&amp;order=desc&amp;page=1&amp;per_page=10
</code></pre>
<p>PFA</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/b/7b65a0356f26a04a005dd7cb43c33831ade3b373.png" data-download-href="/uploads/short-url/hBCsAJPePnNB0u3T5uPuMrDp8VZ.png?dl=1" title="Screenshot 2021-06-07 at 7.04.38 PM" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/7281166/181086547-fd017ebc-ebf7-44b0-87b4-473ee077aba7.png" alt="Screenshot 2021-06-07 at 7.04.38 PM" data-base62-sha1="hBCsAJPePnNB0u3T5uPuMrDp8VZ" width="690" height="472" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/b/7b65a0356f26a04a005dd7cb43c33831ade3b373_2_690x472.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/b/7b65a0356f26a04a005dd7cb43c33831ade3b373_2_1035x708.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/7/b/7b65a0356f26a04a005dd7cb43c33831ade3b373_2_1380x944.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/b/7b65a0356f26a04a005dd7cb43c33831ade3b373_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-06-07 at 7.04.38 PM</span><span class="informations">2122×1452 317 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by gabrieldeoliveiraestevam** [comment]
<p>@ <a href="https://github.community/u/parithiban">parithiban</a></p>
<p>I need to put the same question in both the query and the sort. That is, I need to put.</p>
<p>pushed:&gt;=X&amp;sort=pushed</p>
<p>or</p>
<p>updated:&gt;=X&amp;sort=updated</p>
<p>However, I am not able to do this.</p>
<p>My idea is as follows. API limits 1000 records per request correct? then, after going through the 10 pages with 100 records each. I’m going to lose the information from the last record (updated OR pushed) and do a new search with the timestamp of that last record.</p>

**Reply by YUEchn** [reply]
May I ask if the issue has been resolved? Is the pushed field equal to the updated?
